From ffeb595afd87a8eed1b64c9a815b1e458ca2b734 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:51 -0400
Subject: [tools] perf/thread: Move comm_list check into function

Message-id: <1408953210-26343-44-git-send-email-jolsa@redhat.com>
Patchwork-id: 88751
O-Subject: [PATCH RHEL7.1 BZ1133083 043/322] perf thread: Move comm_list check into function
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 1f3878c11ca8c9f2d9545ec6e1e6bb5386e0565f
Author: David Ahern <dsahern@gmail.com>
Date: Mon Nov 18 13:32:47 2013 -0700
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index bb788c1..c77814b 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -732,8 +732,7 @@ int perf_event__preprocess_sample(const union perf_event *event,
  if (thread == NULL)
   return -1;
 
- if (symbol_conf.comm_list &&
-     !strlist__has_entry(symbol_conf.comm_list, thread__comm_str(thread)))
+ if (thread__is_filtered(thread))
   goto out_filtered;
 
  dump_printf(" ... thread: %s:%d\n", thread__comm_str(thread), thread->tid);
diff --git a/tools/perf/util/thread.h b/tools/perf/util/thread.h
index 897c1b2..5b856bf 100644
--- a/tools/perf/util/thread.h
+++ b/tools/perf/util/thread.h
@@ -6,6 +6,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include "symbol.h"
+#include <strlist.h>
 
 struct thread {
  union {
@@ -66,4 +67,15 @@ static inline void thread__set_priv(struct thread *thread, void *p)
 {
  thread->priv = p;
 }
+
+static inline bool thread__is_filtered(struct thread *thread)
+{
+ if (symbol_conf.comm_list &&
+     !strlist__has_entry(symbol_conf.comm_list, thread__comm_str(thread))) {
+  return true;
+ }
+
+ return false;
+}
+
 #endif /* __PERF_THREAD_H */
-- 
1.7.1